src/kw_ssh.sh

```diff
@@ -112,6 +125,137 @@ function run_command_in_the_remote()
   cmd_manager "$flag" "$ssh_compose"
 }
 
+# This function transfers a local file or directory to a
+# specified remote destination or gets a file or directory
+# from the specified remote path and saves it to the local
+# machine at the specified path.
+#
+# @transfer_path: Path to the file or directory to transfer.
+# This can be either a local path or a remote path, depending
+# on the transfer type.
+#
+# @flag: Expecting a flag, by default, cmd_manager does not
+# expects flags and always show the command. For more details
+# see the function `cmd_manager` in `src/lib/kwlib.sh`.
+#
+# @transfer_type: Transfer type, either 'send' or 'get'.
+function ssh_transfer_file()
+{
+  local flag="$1"
+  local transfer_path="$2"
+  local transfer_type="$3"
+  local user
+  local port
+  local remote
+  local remote_file
+  local remote_file_host
+  local transfer_path
+  local to_path
+  local cmd
+  local ssh_compose
+  local filename
+  local filename_path
+  local exist_path
+  local rsync_options
+
+  flag=${flag:-'SILENT'}
+
+  user="${remote_parameters['REMOTE_USER']}"
+  port="${remote_parameters['REMOTE_PORT']}"
+  remote="${remote_parameters['REMOTE_IP']}"
+  remote_file="${remote_parameters['REMOTE_FILE']}"
+  remote_file_host="${remote_parameters['REMOTE_FILE_HOST']}"
+
+  to_path=${options_values['TO']:-'.'}
+
+  ssh_compose=$(handle_ssh)
+  filename=$(basename "$transfer_path")
+  filename_path=$(printf "%s/%s" "$to_path" "$filename" | tr --squeeze-repeats '/')
+  exist_path="$to_path"
+
+  case "$transfer_type" in
+    'send')
+
+      cmd_manager "$flag" "${ssh_compose} \"[[ -f '${to_path}/${filename}' ]]\""
+
+      if [[ "$?" -eq 0 ]]; then
+        exist_path="$filename_path"
+      fi
+
+      if [[ -n "$exist_path" ]]; then
+        complain "The '$exist_path' file already exists on the remote machine."
+
+        if [[ $(ask_yN 'Do you wish to overwrite it?') =~ '0' ]]; then
+          exit 125 # ECANCELED
+        fi
+      fi
+
+      if [[ ! -e "$transfer_path" ]]; then
+        complain "The file or directory '${transfer_path}' does not exist."
+        exit 2 # ENOENT
+      fi
+
+      cmd="rsync -avzq ${transfer_path} -e "
+      rsync_options="'ssh -p ${port}' ${user}@${remote}:${to_path}"
+
+      if [[ -f "$remote_file" ]]; then
+        rsync_options="'ssh -F ${remote_file}' ${remote_file_host}:${to_path}"
+      fi
+
+      cmd+="$rsync_options"
+
+      cmd_manager "$flag" "$cmd"
+
+      if [[ "$?" -ne 0 ]]; then
+        complain 'An error occurred while uploading the file(s).'
+        exit 22 # EINVAL
+      fi
```

Comment:
I don't think you want to return 22 since rsync has its own set of returns. Actually, rsync has detailed documentation about the return type.